spec/datadog/appsec/component_spec.rb

```diff
@@ -0,0 +1,46 @@
+# typed: ignore
+
+require 'datadog/appsec/spec_helper'
+require 'datadog/appsec/component'
+
+RSpec.describe Datadog::AppSec::Component do
+  describe '.build_appsec_component' do
+    context 'when appsec is enabled' do
+      it 'returns a Datadog::AppSec::Component instance' do
+        Datadog.configuration.appsec.enabled = true
+        component = described_class.build_appsec_component(Datadog.configuration)
+        expect(component).to be_a(described_class)
+      end
+    end
+
+    context 'when appsec is not enabled' do
+      it 'returns nil' do
+        Datadog.configuration.appsec.enabled = false
```

Comment:
I know you adopted this after my previous suggestion, but be careful about solving it in this way because this is state that gets carried over to other tests.

Tests that change the global configuration should make sure to call 
`Datadog.configuration.reset!` at the end (you'll see this in other places in the codebase where we do this.

Although it's kinda uglier, I do still favor my suggestion from https://github.com/DataDog/dd-trace-rb/pull/2632#discussion_r1114180817 as at least that doesn't leave any lingering state, but I'll leave it up to you to decide which approach to better use (just make sure no state gets left! :D).

Reply:
Sounds good

Reply:
We should probably add some tracking for this kind of pattern. I have seen in other tests thats why I used it.